README.md

```diff
@@ -0,0 +1,131 @@
+# What is the Application?
+
+A simple REST API web service consisting of two microservices. The application includes a Java web application that exposes a REST API, which communicates with an Express application.
+
+## Endpoints
+
+### Node Endpoints
+
+1. **GET greeting**
+
+    - **Request:**
+        - Method: `GET`
+        - URL: `/greeting`
+        - Example Request:
+          ```
+          GET /greeting
+          ```
+
+    - **Response:**
+        - Status Code: `200 OK`
+        - Example Response:
+          ```
+          { 
+            "text": "Hello World"
+          }
+          ```
+
+2. **POST process-greeting-node**
+
+    - **Request:**
+        - Method: `POST`
+        - URL: `/process-greeting-node`
+        - Example Request:
+          ```
+          POST /process-greeting-node
+          Content-Type: application/json
+
+          {
+            "name": "Hello Ali"
+          }
+          ```
+
+    - **Response:**
+        - Status Code: `200 OK`
+        - Example Response:
+          ```
+          {  
+            "name": "Hello from Ali"
+          }
+          ```
+
+### Dropwizard Endpoints
+
+1. **GET greeting**
+
+    - **Request:**
+        - Method: `GET`
+        - URL: `/greeting`
+        - Example Request:
+          ```
+          GET /greeting
+          ```
+
+    - **Response:**
+        - Status Code: `200 OK`
+        - Example Response:
+          ```
+          {
+            "text": "hello world"
+          }
+          ```
+
+2. **POST process-greeting**
+
+    - **Request:**
+        - Method: `POST`
+        - URL: `/process-greeting`
+        - Example Request:
+          ```
+          POST /process-greeting
+          Content-Type: application/json
+
+          {
+            "name": "Hello from John"
+          }
+          ```
+
+    - **Response:**
+        - Status Code: `200 OK`
+        - Example Response:
+          ```
+          { 
+            "name": "Hello from John"
+          }
+          ```
+
+## How the Application Works
+
+- The `HelloWorldResource` handles HTTP requests and forwards them to the appropriate endpoints in the Node.js service using JAX-RS.
+- Responses from the Node.js service are then processed by the `HelloWorldResource` and sent back to the client.
+- JAX-RS is the JavaEE API that provides API annotations like `@GET`, `@POST`, etc.
+- Jersey implements the JAX-RS API and handles the routing of HTTP requests to the appropriate methods.
+- Jackson is used for deserializing `app.yml` into `AppConfiguration`.
+- Jackson is also used for deserializing the requests and responses to/from Dropwizard.
+ ### How the node app works ###
+
+- The `paths.js` file defines our paths for the greetings.
+- The `routes.js` file has the routes for the get and post requests.
+- The logic of both get and post requests are extracted to the` greeting.controller.js`.
```

Comment:
Just mention that the functions responsible for processing requests are contained in `greeting.controller.js`